spec/tdd_solver_spec.rb

```diff
@@ -0,0 +1,42 @@
+require_relative '../solver'
+
+describe Solver do
+  before(:each) do
+    @solver = Solver.new
+  end
+
+  context 'testing the factorial method' do
+    it 'returns 1 when given 0' do
+      expect(@solver.factorial(0)).to eql 1
+    end
+
+    it 'returns 120 when given 5' do
+      expect(@solver.factorial(5)).to eql 120
+    end
+  end
+
+  context 'testing the reverse method' do
+    it 'returns "olleh" when given "hello"' do
+      expect(@solver.reverse('hello')).to eql 'olleh'
+    end
+  end
+
+  context 'testing the fizzbuzz method' do
+    it 'returns "fizz" when divisible by 3' do
+      expect(@solver.fizzbuzz(6)).to eql 'fizz'
+    end
+
+    it 'returns "buzz" when divisible by 5' do
+      expect(@solver.fizzbuzz(10)).to eql 'buzz'
+    end
+
+    it 'returns "fizzbuzz" when divisible by 3 and 5' do
+      expect(@solver.fizzbuzz(15)).to eql 'fizzbuzz'
+    end
```

Comment:
- [ ]In testing the fizzbuzz method, you have missed one test case: `Any other case, return N as a string (e.g. say N is 7 then return "7").` that is **N is not divisible by 3 or 5**

Reply:
Thanks! The test case have been added